File: 12_11_2022/increasing_substring/increasing_substring.py
def MaxLengthOfIncreasingSubstr(arr, n):
    arr.append(-1)
    cur_len, max_len = 1, 1
    for i in range(len(arr) - 1):
        if arr[i] < arr[i + 1]:
            cur_len += 1
            max_len = max(cur_len, max_len)
        else:
            cur_len = 1
    return max_len
    

def main():
    arr_len = int(input())
    arr  = list(map(int, input().split()))
    # for i in range(arr_len):
    #     arr.append(int(input()))
    print(MaxLengthOfIncreasingSubstr(arr, arr_len))     
        
if __name__ == '__main__':
    main()

File: 12_11_2022/increasing_substring/increasing_substring.cpp
#include <bits/stdc++.h>

int main() {
    int n, prev_el = 0, max_len = 1, cur_len = 1;
    std::cin >> n >> prev_el;

    int cur_el;
    
    for (int i = 0; i < n - 1; ++i) {
        std::cin >> cur_el;
        if (prev_el < cur_el) {
            ++cur_len;
            max_len = std::max(cur_len, max_len);
        }
        else { cur_len = 1; }
        prev_el = cur_el;
    }
    std::cout << max_len << "\n";
    return 0;
}